pkgs/firehose/lib/health.dart

```diff
@@ -0,0 +1,261 @@
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// ignore_for_file: always_declare_return_types
+
+import 'dart:io';
+
+import 'package:firehose/firehose.dart';
+import 'package:firehose/src/repo.dart';
+import 'package:path/path.dart' as path;
+
+import 'src/github.dart';
+import 'src/utils.dart';
+
+const String _botSuffix = '[bot]';
+
+const String _githubActionsUser = 'github-actions[bot]';
+
+const String _publishBotTag2 = '### Package publish validation';
+
+const String _licenseBotTag = '### License Headers';
+
+const String _changelogBotTag = '### Changelog Entry';
+
+const String _prHealthTag = '## PR Health';
+
+class Health {
+  final Directory directory;
+
+  Health(this.directory);
+
+  Future<void> healthCheck(List args) async {
+    var github = Github();
+
+    // Do basic validation of our expected env var.
+    if (!expectEnv(github.githubAuthToken, 'GITHUB_TOKEN')) return;
+    if (!expectEnv(github.repoSlug, 'GITHUB_REPOSITORY')) return;
+    if (!expectEnv(github.issueNumber, 'ISSUE_NUMBER')) return;
+    if (!expectEnv(github.sha, 'GITHUB_SHA')) return;
+
+    if ((github.actor ?? '').endsWith(_botSuffix)) {
+      print('Skipping package validation for ${github.actor} PRs.');
+      return;
+    }
+
+    print('Start health check for the checks $args');
+    var checks = [
+      if (args.contains('version') &&
+          !github.prLabels.contains('skip-validate-check'))
+        validateCheck,
+      if (args.contains('license') &&
+          !github.prLabels.contains('skip-license-check'))
+        licenseCheck,
+      if (args.contains('changelog') &&
+          !github.prLabels.contains('skip-changelog-check'))
+        changelogCheck,
+    ];
+
+    var checked =
+        await Future.wait(checks.map((check) => check(github)).toList());
+    await writeInComment(github, checked);
+
+    github.close();
+  }
+
+  Future<HealthCheckResult> validateCheck(Github github) async {
+    var results = await Firehose(directory).verify(github);
+
+    var markdownTable = '''
+| Package | Version | Status | Publish tag (post-merge) |
+| :--- | ---: | :--- | ---: |
+${results.describeAsMarkdown}
+
+Documentation at https://github.com/dart-lang/ecosystem/wiki/Publishing-automation.
+    ''';
+
+    return HealthCheckResult(
+      _publishBotTag2,
+      results.severity,
+      markdownTable,
+    );
+  }
+
+  Future<HealthCheckResult> licenseCheck(Github github) async {
+    final license = '''
+// Copyright (c) ${DateTime.now().year}, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.''';
+
+    var filePaths = await _getFilesWithoutLicenses(github);
+
+    var markdownResult = '''
+```
+$license
+```
+
+| Files |
+| :--- |
+${filePaths.map((e) => '|$e|').join('\n')}
```

Comment:
If no files w/o missing copyright headers, perhaps still emit a row for the table?

```
| _no missing headers_ |
```

?